packages/sit-onyx/src/styles/grid.scss

```
@@ -126,13 +135,31 @@ $sidebar-breakpoints: (
     }
   }
 
+  /**
+ * Initializes the grid layout on the `.page__main` element. The layout will adapt
+ * to the width of this wrapper accordingly.
+ */
+  .onyx-page__main {
+    container-type: inline-size;
+  }
+
   @each $name, $maxVariant, $columns, $gutter, $margin in $gridVariants {
     @if $name == "2xs" {
-      @include generate-grid-properties($columns, $gutter, $margin);
+      @include generate-grid-properties(
+        $columns,
+        $gutter,
+        $margin,
+        ".onyx-grid, .onyx-grid-layout, .onyx-grid-container"
+      );
```

It would be preferable to move the selector around the `@include generate-grid-properties`, instead of passing it as a parameter. That way it is closer to CSS and only parameters are passed that are directly used by the function

```suggestion
      @include generate-grid-properties(
        $columns,
        $gutter,
        $margin,
        ".onyx-grid, .onyx-grid-layout, .onyx-grid-container"
      );
```